src/OrchardCore.Modules/OrchardCore.Workflows/Scripting/JavaScriptWorkflowScriptEvaluator.cs

```diff
@@ -37,6 +37,14 @@ public async Task<T> EvaluateAsync<T>(WorkflowExpression<T> expression, Workflow
         await _workflowContextHandlers.InvokeAsync((h, expressionContext) => h.EvaluatingScriptAsync(expressionContext), expressionContext, _logger);
 
         var methodProviders = scopedMethodProviders.Concat(expressionContext.ScopedMethodProviders);
-        return (T)_scriptingManager.Evaluate(directive, null, null, methodProviders);
+
+        try
+        {
+            // Some types cannot be cast (e.g., null to bool), so we need to catch the exception and return the default value.
+            return (T)_scriptingManager.Evaluate(directive, null, null, methodProviders);
+        }
+        catch { }
```

Comment:
Swallow the exception is not a good practice, at least log a warning message if it's possible

Reply:
done